### PR TITLE
Setup MirageServer and RTKQuery

### PR DESCRIPTION
**Pull request by faisalusmanprixite**

### CODE REVIEW

File: src/components/common/Presentational/TrailList/TrailList.tsx [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
 const TrailList = () => {
   const navigate = useNavigate();
+  const [trails, setTrails] = useState<Array<ITrailData>>([]);
   const { trailList } = useAppSelector((state) => state.appData);
+  useEffect(() => {
+    retrieveTrails();
+  }, []);
+
+  const retrieveTrails = () => {
+    TrailsDataService.getAll()
```
Comment (getalihassan) [inline src/components/common/Presentational/TrailList/TrailList.tsx:20]: please avoid chainning method, convert it into promise style with async/await. and avoid this approach as I have told you many times

**Review by getalihassan — CHANGES_REQUESTED** [review]

**Review by getalihassan — CHANGES_REQUESTED** [review]

File: src/services/TrailsService.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import http from 'src/http-common';
```
Comment (getalihassan) [inline src/services/TrailsService.ts:1]: this file will get refactored please connect with Abdullah

**Review by getalihassan — CHANGES_REQUESTED** [review]

**Comment by getalihassan** [comment]
@faisalusmanprixite change the name of branch and name of PR

File: package.json [hunk 3]
```diff
@@ -17,6 +17,7 @@
     "@types/react": "^18.0.9",
     "@types/react-dom": "^18.0.4",
     "@types/react-router-dom": "^5.3.3",
+    "axios": "^0.27.2",
```
Comment (getalihassan) [inline package.json:20]: remove this as we are not using axsio now

File: src/components/common/Presentational/TrailList/TrailList.tsx [hunk 4]
```diff
@@ -3,15 +3,17 @@ import Typography from '@mui/material/Typography';
...
 import { useNavigate } from 'react-router-dom';
-import { useAppSelector } from 'src/store/hooks';
+import { useGetAllTrailsQuery } from 'src/store/reducers/api';
 import 'src/components/common/Presentational/TrailList/TrailList.scss';
 
 const TrailList = () => {
   const navigate = useNavigate();
-  const { trailList } = useAppSelector((state) => state.appData);
+  const { data: trails = [] } = useGetAllTrailsQuery({});
+  console.log('traildata', trails);
```
Comment (getalihassan) [inline src/components/common/Presentational/TrailList/TrailList.tsx:12]: remove console

**Review by getalihassan — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (faisalusmanprixite) [reply package.json:20]: ok

File: src/components/common/Presentational/TrailList/TrailList.tsx (hunk 4 above)
Comment (faisalusmanprixite) [reply src/components/common/Presentational/TrailList/TrailList.tsx:12]: ok

**Pull request merged**